src/parkapi_sources/converters/apcoa/validators.py

```diff
@@ -0,0 +1,194 @@
+    CarparkPhotoURL2: Optional[str] = Noneable(UrlValidator())
+    CarparkPhotoURL3: Optional[str] = Noneable(UrlValidator())
+    CarparkPhotoURL4: Optional[str] = Noneable(UrlValidator())
+
+
+@validataclass
+class ApcoaIndicativeTariffInput:
+    MinPrefix: Optional[str] = Noneable(StringValidator())
+    MinValue: Optional[Decimal] = Noneable(NumericValidator())
+    MinSuffix: Optional[str] = Noneable(StringValidator())
+    MaxPrefix: Optional[str] = Noneable(StringValidator())
+    MaxValue: Optional[Decimal] = Noneable(NumericValidator())
+    MaxSuffix: Optional[str] = Noneable(StringValidator())
+    Currency: Optional[str] = Noneable(StringValidator())
+    CurrencyCode: Optional[str] = Noneable(StringValidator())
+    TaxRate: Optional[Decimal] = Noneable(NumericValidator())
+
+
+@validataclass
+class ApcoaParkingSiteInput:
+    CarParkId: int = IntegerValidator(allow_strings=True)
+    CarparkLongName: Optional[str] = Noneable(StringValidator())
+    CarparkShortName: Optional[str] = Noneable(StringValidator())
+    CarParkWebsiteURL: Optional[str] = Noneable(UrlValidator())
+    CarParkPhotoURLs: Optional[ApcoaCarparkPhotoURLInput] = Noneable(DataclassValidator(ApcoaCarparkPhotoURLInput))
+    CarparkType: ApcoaCarparkTypeNameInput = DataclassValidator(ApcoaCarparkTypeNameInput)
+    Address: ApcoaAdressInput = DataclassValidator(ApcoaAdressInput)
+    NavigationLocations: list[ApcoaNavigationLocationsInput] = ListValidator(DataclassValidator(ApcoaNavigationLocationsInput))
+    Spaces: list[ApcoaParkingSpaceInput] = ListValidator(DataclassValidator(ApcoaParkingSpaceInput))
+    OpeningHours: list[ApcoaOpeningHoursInput] = ListValidator(DataclassValidator(ApcoaOpeningHoursInput))
+    LastModifiedDateTime: datetime = SpacedDateTimeValidator(
+        local_timezone=ZoneInfo('Europe/Berlin'),
+        target_timezone=timezone.utc,
+    )
+    IndicativeTariff: Optional[ApcoaIndicativeTariffInput] = Noneable(DataclassValidator(ApcoaIndicativeTariffInput))
+
+    # TODO: ignored multiple attributes which do not matter so far
+
+    def __post_init__(self):
+        for capacity in self.Spaces:
+            if capacity.Type == ApcoaParkingSpaceType.TOTAL_SPACES:
+                return
+        # If no capacity with type PARKING was found, we miss the capacity and therefore throw a validation error
+        raise ValidationError(reason='Missing parking spaces capacity')
+
+    def get_osm_opening_hours(self) -> str:
+        apcoa_opening_times: dict = defaultdict(list)
+        for opening_hours_input in self.OpeningHours:
+            opening_time = opening_hours_input.OpeningTimes.replace(' ', '')
+            if opening_hours_input.OpeningTimes == '00:00 - 00:00':
+                apcoa_opening_times['24/7'].append(opening_time)
+            if opening_hours_input.Weekday in list(ApcoaOpeningHoursWeekdays)[:5]:
+                apcoa_opening_times['Mo-Fr'].append(opening_time)
```

Comment:
This is quite a good example why you need unittests for this transformation. When you have an input like `Monday: 10:00 - 20:00, Tuesday: 10:00 - 12:00, Saturday: 10:00 - 16:00`, this ends up in `Mo-Fr: 10:00 - 20:00, 10:00 - 12:00; Sa: 10:00 - 16:00`.

Reply:
Added an example unittest testing common transformations